electron/src/updater/updateService.ts

```diff
@@ -0,0 +1,206 @@
+"use strict";
```

Comment:
The gut of the update logic. It is responsible for downloading and caching the release manifest, interpreting it, and emitting update events to the shell (which in turns pipes these events to the webapp, which can then display dialogs to the user).